gplearn/genetic.py

```diff
@@ -471,7 +471,10 @@ def fit(self, X, y, sample_weight=None):
 
         if isinstance(self, RegressorMixin):
             # Find the best individual in the final generation
-            self._program = self._programs[-1][np.argmin(fitness)]
+            if self._metric.greater_is_better:
+                self._program = self._programs[-1][np.argmax(fitness)]
+            else:
+                self._program = self._programs[-1][np.argmin(fitness)]
 
         if isinstance(self, TransformerMixin):
             # Find the best individuals in the final generation
```

Comment:
Can we expand the PR to cover the transformer as well? I think that the `hall_of_fame = fitness.argsort()[:self.hall_of_fame]` should be the place to do this. Should be adequate to cut the results in the opposite direction for the bottom end using `hall_of_fame = fitness.argsort()[self.population_size - self.hall_of_fame:]`